pex/common.py

```diff
@@ -276,7 +276,10 @@ def write(self, data, dst, label=None, mode='wb'):
     self._tag(dst, label)
     self._ensure_parent(dst)
     with open(os.path.join(self.chroot, dst), mode) as wp:
-      wp.write(data)
+      try:
+        wp.write(data)
+      except TypeError:
+        wp.write(bytes(data, 'UTF-8'))
```

Comment:
this doesn't seem right.  it should not be the responsibility of write() to do the encoding/decoding.  it should be the responsibility of the caller.